src/ILCompiler/src/Program.cs

```diff
@@ -437,6 +437,12 @@ private int Run(string[] args)
                 }
             }
 
+            // R2R field layout needs compilation group information; this is the earliest point when we can set it
+            if (typeSystemContext is ReadyToRunCompilerContext r2rContext)
+            {
+                r2rContext.SetCompilationGroup(compilationGroup);
```

Comment:
Could you move this to e.g. ReadyToRunCodegenCompilationBuilder constructor?

Reply:
Moved in 6th commit.